gemfirexd/client/src/main/java/io/snappydata/jdbc/ClientPoolDriver.java

```diff
@@ -19,123 +19,151 @@
+  public boolean jdbcCompliant() {
+    return ClientConfiguration.jdbcCompliant;
+  }
+
+  @Override
+  public Logger getParentLogger() throws SQLFeatureNotSupportedException {
+    throw new AssertionError("should be overridden in JDBC 4.1");
+  }
+
+  @Override
+  public DriverPropertyInfo[] getPropertyInfo(String url, Properties properties)
+      throws SQLException {
+
+    java.sql.DriverPropertyInfo driverPropertyInfo[] = new java.sql.DriverPropertyInfo[2];
+
+    // If there are no properties set already,
+    // then create a dummy properties just to make the calls go thru.
+    if (properties == null) {
+      properties = new java.util.Properties();
     }
 
-    @Override
-    public Connection connect(String url, Properties properties) throws SQLException {
-
-        if (!acceptsURL(url)) {
-            return null;
-        }
-
-        properties = (properties == null) ? new Properties() : properties;
-        String clientDriverURL = url.toLowerCase().replace("pool:", "");
-        properties.setProperty(TomcatConnectionPool.PoolProps.URL.key, clientDriverURL);
-        properties.setProperty(TomcatConnectionPool.PoolProps.DRIVER_NAME.key, ClientDriver.class.getName());
-
-        TomcatConnectionPool p = poolMap.get(properties);
-        if (p != null) {
-            Connection conn = p.getConnection();
-            CURRENT_CONNECTION.set(conn);
-            return conn;
-        } else {
-            TomcatConnectionPool pool;
-            synchronized (ClientPoolDriver.class) {
-                pool = poolMap.get(properties);
-                if (pool == null) {
-                    pool = new TomcatConnectionPool(properties);
-                    poolMap.put(properties, pool);
-                }
-            }
-            Connection conn = pool.getConnection();
-            CURRENT_CONNECTION.set(conn);
-            return conn;
-        }
+    boolean isUserNameAttribute = false;
+    String userName = properties.getProperty(Attribute.USERNAME_ATTR);
+    if (userName == null) {
+      userName = properties.getProperty(Attribute.USERNAME_ALT_ATTR);
+      if (userName != null) {
+        isUserNameAttribute = true;
+      }
     }
-}
+
+    driverPropertyInfo[0] = new java.sql.DriverPropertyInfo(
+        isUserNameAttribute ? Attribute.USERNAME_ALT_ATTR
+            : Attribute.USERNAME_ATTR, userName);
+
+    driverPropertyInfo[1] =
+        new java.sql.DriverPropertyInfo(Attribute.PASSWORD_ATTR,
+            properties.getProperty(Attribute.PASSWORD_ATTR));
+
+    driverPropertyInfo[0].description =
+        SqlException.getMessageUtil().getTextMessage(
+            MessageId.CONN_USERNAME_DESCRIPTION);
+    driverPropertyInfo[1].description =
+        SqlException.getMessageUtil().getTextMessage(
+            MessageId.CONN_PASSWORD_DESCRIPTION);
+
+    driverPropertyInfo[0].required = true;
+    driverPropertyInfo[1].required = false; // depending on the security mechanism
+
+    return driverPropertyInfo;
```

Comment:
This method is identical to that in ClientDRDADRiver. It doesn't need any member fields so you could refactor that into a static method in ClientDRDADriver and invoke from here too.

Reply:
Yes, refactored it..